backend/src/parser/TheParser.ts

```diff
@@ -227,6 +236,26 @@ export default class TheParser {
         return false;
     }
 
+    processInternalUrl(url: string, text?: string) {
+        const internalUrlPattern = new RegExp(`^(https?://${this.parserConfig.siteDomain})/(?:s/([^/]+)/)?p(\\d+)(?:#(\\d+))?$`, 'i');
+
+        const match = url.match(internalUrlPattern);
+        const site = match && match[2] || null;
+        if (match && (!site || !joiSite.validate(site).error)) {
+            const host = match[1];
+            const postId = match[3];
+            const commentId = match[4] || null;
+
+            const origUrl = `${host}/${site ? 's/' + site + '/' : ''}p${postId}${commentId ? '#' + commentId : ''}`;
+            const expandButton = `<span role="button" class="expand-button i i-expand" data-post-id="${postId}"${commentId ? ' data-comment-id="' + commentId + '"' : ''}></span>`;
```

Comment:
I think we shouldn't be using `expand` icon on different purpose. May be hyperlink icon?
add tooltip for better UX
```suggestion
            const expandButton = `<span role="button" class="expand-button i i-expand" title="А что там?" data-post-id="${postId}"${commentId ? ' data-comment-id="' + commentId + '"' : ''}></span>`;
```

Reply:
I kinda agree that slightly different icon would be better, but it shouldn't be completely different, as both actions share the similar "expand" part.

@ArtCraft, do you think you can come up with a better icon, or is the current one fine?
<img width="330" alt="image" src="https://github.com/spaceshelter/orbitar/assets/2865203/4a8ad19e-ab8c-4489-b4da-21908c6a4021">